src/components/Notification/Notification.tsx

```diff
@@ -2,43 +2,68 @@ import * as moment from "moment";
 import * as React from "react";
 import * as css from "./Notification.scss";
 import { IOperation, OperationsStatus } from "reducers/operations";
+import classNames = require("classnames");
 
 interface IProps {
   operation: IOperation;
 
   close: () => any;
 }
 
-export default class Notification extends React.Component<IProps, null> {
+export default class Notification extends React.Component<IProps, { minimized: boolean }> {
+  constructor(props: any) {
+    super(props);
+    this.state = { minimized: false };
+  }
+
   public handleClose = (e: any) => {
-    const {close} = this.props;
+    const { close } = this.props;
     close();
   }
 
   public render() {
     const { operation } = this.props;
     const { message, status, totalSteps, timestamp, step } = operation;
+    const { minimized } = this.state;
+
+    const transactionClass = classNames({
+      [css.pendingTransaction]: true,
+      [css.clearfix]: true,
+      [css.pending]: status === OperationsStatus.Pending,
+      [css.error]: status === OperationsStatus.Failure,
+      [css.success]: status === OperationsStatus.Success,
+      [css.minimized]: status === OperationsStatus.Pending && minimized,
+    });
 
     return (
-      <div className={css.notification}>
-        <span className={css.message}>{message}</span>
-        <br/>
-        <span className={css.message}>status: {status}</span>
-        <br/>
-        { totalSteps ?
-          (totalSteps === 1 ?
-            '' :
-            <span>step {step} out of {totalSteps}</span>
-          ) :
-          <span>step {step}</span> }
-        <br/>
-        <small className={css.timestamp}>{moment(timestamp).fromNow()}</small>
-        { operation.status !== OperationsStatus.Pending ? <img
-          src="/assets/images/Icon/Close.svg"
-          alt="close"
-          className={css.close}
-          onClick={this.handleClose}/>
-          : ''}
+      <div className={transactionClass} onClick={() => status === OperationsStatus.Pending && minimized ? this.setState({minimized: false}) : null }>
```

Comment:
lets pull this out into a separate function

Reply:
updated: https://github.com/daostack/alchemy/pull/52/commits/191f7d3b870fba072c530d61301349f4f3f7ff94